PathSumII.java

```diff
@@ -0,0 +1,33 @@
+// Time Complexity: O(n)
+// Space Complexity with recursive stack: O(n +maxDepth)
+// Space Complexity without recursive stack: O(n)
+
+class Solution {
+
+    public List<List<Integer>> pathSum(TreeNode root, int sum) {
+        List<List<Integer>> res = new ArrayList<>();
+        helper(root, sum, 0, new ArrayList<>(), res);
+        return res;
+    }
+
+    private void helper(TreeNode node, int target, int curSum, List<Integer> list, List<List<Integer>> res) {
+        if(node == null) {
+            return;
+        }
+
+        list.add(node.val);
+
+        if(node.left == null && node.right == null) {
+            if(curSum + node.val == target) {
+                res.add(new ArrayList<Integer>(list));
```

Comment:
Leetcode says the time complexity for this question is O(n^2). The time taken to copy this list is O(logn) (height of the tree and O(n) worst case.